cmake/bintools/gnu/ld_version.cmake

```diff
@@ -0,0 +1,31 @@
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
this file should be converted into a `Findld.cmake` and placed in this folder:
https://github.com/zephyrproject-rtos/zephyr/tree/main/cmake/modules

that will allow us to call:
```
find_package(ld)
```
instead of `include(${ZEPHYR_BASE}/cmake/bintools/gnu/ld_version.cmake)`.

That will also follow the common CMake practice of providing a `<package_name>_VERSION` setting, as well as `<package_name>_FOUND`.
One can even do a `find_package(ld REQUIRED)` to ensure `ld` is found, and fail if not.

To consider:
- is `ld` the best name, or should it be `Gnuld` /  `binutils_ld`, or similar ?
I'm open to good naming suggestions on this.

**Note**: making a `Findld.cmake` will also allow us to update this code:
https://github.com/zephyrproject-rtos/zephyr/blob/0c9ca9aefaa1e6a54dfc23ce4e74bcd3c58dd3f3/cmake/linker/ld/target.cmake#L4-L13
to simply use `find_package(ld REQUIRED)`

Reply:
> is ld the best name, or should it be Gnuld / binutils_ld, or similar ?

It could be called `ld.bfd`, that's how both Debian and Fedora call it. I think many more people understand "GNU ld" however. `ld.bfd` would not be ambiguous but many people would have to google it.

```
$ md5sum /usr/bin/ld{,.bfd}
9b6cfc52ea4aa58adf87c961c4ae5a9d  /usr/bin/ld
9b6cfc52ea4aa58adf87c961c4ae5a9d  /usr/bin/ld.bfd
```

GNU ld should really not be called just `ld` because that's too generic and ambiguous. For instance the native, Mach-O linker on macOS is also called `ld`.

lld - LLVM
ld.gold
ld / ld.bfd - GNU ld
ld - Mach-O linker on macOS

Reply:
Let me rework the PR to support `find_package()`.